### DISCUSSION TITLE
RFC: Implement P2P for files

### DISCUSSION THREAD

**Discussion by filiptronicek** [interclip/interclip-next#122] [Ideas]
For big files (1GB+) and for confidential files which should be kept secret, it's reasonable to assume that users would want a way of storing files without them ever being stored on a server. This will most likely be a pain to setup, but it is something I really believe in. 

We could implement the logic behind https://file.pizza/, but I couldn't get it to work.

**Comment by filiptronicek** [comment]
We could also implement https://github.com/webtorrent/webtorrent, it is reliable and could be nice for some files, exactly like https://github.com/webtorrent/instant.io does it.

**Comment by bigint** [comment]
So the decentralised files don’t go down?

**Reply by filiptronicek** [reply]
They certainly do, they do down as soon as all the peers turn off Interclip in their browser.

**Reply by bigint** [reply]
Yup @filiptronicek that's what I exactly asked in https://github.com/interclip/interclip-next/discussions/122#discussioncomment-1668275 😄

**Reply by filiptronicek** [reply]
Yes, the goal here is to have this work in poor network conditions and allow for uploading unlimited file sizes.

**Comment by bigint** [comment]
Who gonna pin my files uploaded in torrent?

**Reply by bigint** [reply]
![image](https://user-images.githubusercontent.com/69431456/142567269-bb1826c8-93d6-4850-b07e-ef65d9fe9b50.png)

This happens when it is not pinned 😄 

IPFS is better

**Reply by filiptronicek** [reply]
Bruh man what? IPFS solves 0/2 problems I outlined in the discussion description.

**Reply by filiptronicek** [reply]
It's for the times when both of the devices are open with the file, not a month afterwards. They're temporary

**Reply by bigint** [reply]
woooh if temporary its ok 👍🏻
